src/h3lib/lib/polyfill.c

```diff
@@ -448,20 +450,31 @@ void iterStepPolygonCompact(IterCellsPolygonCompact *iter) {
             }
             if (mode == CONTAINMENT_OVERLAPPING) {
                 // For overlapping, we need to do a quick check to determine
-                // whether the polygon is wholly contained by the cell. We check
-                // the first polygon vertex, which if it is contained could also
-                // mean we simply intersect.
-                H3Index polygonCell;
-                H3Error polygonCellErr = H3_EXPORT(latLngToCell)(
-                    &(iter->_polygon->geoloop.verts[0]), cellRes, &polygonCell);
-                if (polygonCellErr != E_SUCCESS) {
-                    iterErrorPolygonCompact(iter, polygonCellErr);
-                    return;
-                }
-                if (polygonCell == cell) {
-                    // Set to next output
-                    iter->cell = cell;
-                    return;
+                // whether the polygon is wholly contained by the cell. We
+                // check the first polygon vertex, which if it is contained
+                // could also mean we simply intersect.
+
+                // Deferencing verts[0] is safe because we check numVerts above
+                LatLng firstVertex = iter->_polygon->geoloop.verts[0];
+
+                // We have to check whether the point is in the expected range
+                // first, because out-of-bounds values will yield false
+                // positives with latLngToCell
+                if (bboxContains(&VALID_RANGE_BBOX, &firstVertex)) {
+                    H3Index polygonCell;
+                    H3Error polygonCellErr = H3_EXPORT(latLngToCell)(
+                        &(iter->_polygon->geoloop.verts[0]), cellRes,
+                        &polygonCell);
```

Comment:
this is only an optimization so it does not matter if the first vertex would actually overlap the cell or if other parts of the polygon overlap the cell?

Reply:
This check is here because our other logic didn't catch polygons that were entirely contained by the cell (a situation that only applies to `CONTAINMENT_OVERLAPPING`). This catches that case, and additionally catches other cases that would be caught by the more expensive line-intersection check. So it's not an optimization - it's required for the cell-contains-polygon case - but it might cheaply catch other cases as well.

Does that answer the question?

Reply:
Might it be the case that a user passes in a polygon that is entirely contained by a cell, but the first vertex is expressed as over the antimeridian (> M_PI)? In that case, wouldn't it be necessary to check the later vertexes since one of them might be validly in range? Or that the user would expect that single out-of-range vertex to result in latLngToCell?

Reply:
I'm comfortable requiring input for `polygonToCells` to be within valid coordinate range if you want reliable output. Because we use Cartesian logic for most of the checks here, coordinates > M_PI are likely to fail in a number of places. We should definitely guard against bad input causing segfaults or insecure memory access, but I don't feel like we need to try to produce good results for bad input. FWIW I'm pretty sure our previous algo wouldn't properly fill a polygon with out-of-bounds coords either.

Reply:
I am ok with this if it not worse than the previous algorithm.